src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -769,19 +798,20 @@ void ClusterQueueHelper::processPendingContexts(
     // Swap the contexts to process them one by one and also clear the
     // pendingContexts of the queue info: they will be enqueued back, if
     // needed.
-    bsl::vector<OpenQueueContext> contexts(d_allocator_p);
+    bsl::vector<OpenQueueContextSp> contexts(d_allocator_p);
     contexts.swap(queueContext->d_liveQInfo.d_pending);
 
-    for (bsl::vector<OpenQueueContext>::iterator it = contexts.begin();
+    for (bsl::vector<OpenQueueContextSp>::iterator it = contexts.begin();
          it != contexts.end();
          ++it) {
         processOpenQueueRequest(*it);
     }
 }
 
-void ClusterQueueHelper::assignUpstreamSubqueueId(OpenQueueContext* context)
+void ClusterQueueHelper::assignUpstreamSubqueueId(
+    const OpenQueueContextSp& context)
 {
-    QueueLiveState&   info  = context->d_queueContext_p->d_liveQInfo;
+    QueueLiveState&   info  = context->queueContext()->d_liveQInfo;
```

Comment:
Might rely on the checks done before calling this function, or might add them here as well
```suggestion
    // executed by the cluster *DISPATCHER* thread

    // PRECONDITIONS
    BSLS_ASSERT_SAFE(context);
    BSLS_ASSERT_SAFE(context->queueContext());

    QueueLiveState&   info  = context->queueContext()->d_liveQInfo;
```